### ISSUE TITLE
Passenger Install Nginx and Ubuntu 18.10

### ISSUE THREAD

**Issue by mlh758** [phusion/passenger#2104]
**Question 1: What is the problem?**
Following the setup guide listed on the site [here](https://www.phusionpassenger.com/library/install/nginx/install/oss/bionic/)

- What is the expected behavior?
Step 1 completes without error
- What is the actual behavior?
```
The following packages have unmet dependencies:
 libnginx-mod-http-passenger : Depends: nginx-common (= 1.14.0-0ubuntu1) but 1.14.0-0ubuntu2 is to be installed
```
- How can we reproduce it? Please try to provide a sample application (or Virtual Machine) demonstrating the issue. Otherwise, if we can't reproduce it, we might have to ask you a number of followup questions or run certain commands to try and figure out the problem.

Dockerfile content:
```
FROM ubuntu:18.10

RUN apt-get update && \
  # Install packages that our apps will need to run
  apt-get install -y ruby2.5 make ruby-dev gcc zlib1g-dev libaio1 nginx && \
  # Install Passenger
  apt-get install -y dirmngr gnupg apt-transport-https ca-certificates && \
  apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 561F9B9CAC40B2F7 && \
  apt-get install -y apt-transport-https ca-certificates && \
  sh -c 'echo deb https://oss-binaries.phusionpassenger.com/apt/passenger bionic main > /etc/apt/sources.list.d/passenger.list' && \
  apt-get update && \
  apt-get install -y libnginx-mod-http-passenger && \
  apt-get clean

```

Be as detailed as possible in your descriptions, include any logs and stack traces (don't just cut/paste the error, provide some logging before that too).

(if you are requesting a feature instead of reporting an issue, describe here what you have in mind and how it would help you)

Your answer:

**Question 2: Passenger version and integration mode**:
- For example: open source 5.0.26 standalone; enterprise 5.0.21/nginx

Your answer: Whatever comes out of your apt repo

**Question 3: OS or Linux distro, platform (including version)**:
- For example: Debian 8, x86_64 or OS X 10.10 Yosemite, x86_64

Your answer: Ubunutu 18.10

**Question 4: Passenger installation method**:

Your answer:
[ ] RubyGems + Gemfile
[ ] RubyGems, no Gemfile
[x] Phusion APT repo
[ ] Phusion YUM repo
[ ] OS X Homebrew
[ ] source tarball
[ ] Other, please specify:

**Question 5: Your app's programming language (including any version managers) and framework (including versions)**:
- For example: Ruby 2.3.0, RVM, Rails 5.0.0; Node.js 4.0.0 with Express 4.13.4

Your answer: Ruby 2.5.1

**Question 6: Are you using a PaaS and/or containerization? If so which one?**
- For example: Heroku, Amazon Container Services, Docker 1.9 with an image based on passenger-docker

Your answer: Docker

**Question 7: Anything else about your setup that we should know?**

Your answer:

---

We strive for quality and appreciate you taking the time to submit a report! Please note that if you want guaranteed response times and priority issue support we encourage you to join our [enterprise customer](https://www.phusionpassenger.com/enterprise) base. They also provide us with the means to continue our high level of open source support!

**Comment by CamJN** [comment]
We do not support ubuntu 18.10 yet.

**Comment by vladipus** [comment]
I have the same issue with Debian 8 (stretch). Seems like your package depends on the official `nginx-common` package, but I'm using the one comming from NGINX itself and it is called `nginx` (without `-common`).

**Comment by noniq** [comment]
I have a very similar issue with a fresh install of Ubuntu 18.04.1 (Digital Ocean Droplet):

```
The following packages have unmet dependencies:
 libnginx-mod-http-passenger : Depends: nginx-common (= 1.14.0-0ubuntu1) but 1.14.0-0ubuntu1.1 is to be installed
E: Unable to correct problems, you have held broken packages.
```

Is there a workaround for this?

**Comment by FooBarWidget** [comment]
@noniq Yours is a legit problem, though it is a *separate* problem. We'll update the packages soon, but for now you should install Passenger from source.

I've split your issue into a separate ticket: #2122.

**Comment by Myusuf060696** [comment]
I have the same issue
libnginx-mod-http-passenger : Depends: nginx-common (= 1.14.0-0ubuntu1.1) but 1.14.0-0ubuntu1 is to be installed
on ubuntu 18.04.1

**Comment by Malezha** [comment]
@Myusuf060696 you need downgrade nginx to 1.14.0-0ubuntu1 and passenger to 1:5.3.4-1~bionic1

**Comment by cferry322** [comment]
@CamJN - is Ubuntu 18.10 going to be supported sometime soon or would you recommend using a different environment?

**Comment by CamJN** [comment]
@cferry322 Cosmic has been supported since 6.0.1. @shaunakv1 I'm not sure why that's happenining. I just tested a clean ubuntu cosmic image and was able to install nginx and libnginx-mod-http-passenger and wound up with Nginx 1.15.5-0ubuntu2.1 just like the error says you should have.

Perhaps try an `apt clean && apt update` first, and try again?

**Comment by shaunakv1** [comment]
No luck. tried updating the packages. Nginx is still at the same version.

![image](https://user-images.githubusercontent.com/732604/51921979-462ec100-23b6-11e9-89dc-8bc5eeba8e4c.png)

**Comment by CamJN** [comment]
@shaunakv1 you have bionic in your `/etc/apt/sources.list.d/passenger.list` instead of cosmic, that's probably the issue. If you saw that in our docs then that is a bug in our docs it should be cosmic.

**Comment by shaunakv1** [comment]
@CamJN  that was it! changing the `bionic` to `cosmic` in `/etc/apt/sources.list.d/passenger.list` and installing again worked like a charm. 

I was working from the official passenger docs. Those need to be updated! 

Thanks for you help!

**Comment by CamJN** [comment]
@NSLog0 can you post the exact series of steps you took, and the output from `apt-get` that you're seeing?

**Comment by cferry322** [comment]
> @NSLog0 can you post the exact series of steps you took, and the output from `apt-get` that you're seeing?

I was following a guide posted on Medium (obviously that is not your guide, but the steps are about spinning up a Virtual Server on Digital Ocean - 

https://medium.com/ruby-on-rails-web-application-development/how-to-deploy-ruby-on-rails-apps-to-the-internet-production-staging-49efc503c91d.  

It looks like someone from your team mentioned the issue in a blog post here https://blog.phusion.nl/2018/10/03/passenger-5-3-5/  

I am failing on this command 

sudo apt-get install -y libnginx-mod-http-passenger 

I get the failure message

"libnginx-mod-http-passenger : Depends: passenger (= 1:5.1.12-1~zesty1) but it is not going to be installed
Depends: nginx-common (= 1.10.3-1ubuntu3.1) but 1.14.0-0ubuntu1.2 is to be installed"

In the guide I am following, it is the step that is underneath the "Web Server" header.  It seems like from the comments in the blog post that it may not be an operating system issue.  Hope this helps.

![phusion_failingtoinstall_nginxdependency](https://user-images.githubusercontent.com/12903168/52003824-3339f080-248b-11e9-9899-007989e028d7.PNG)

**Comment by voordev** [comment]
Solutions provided NOT working

**Comment by voordev** [comment]
The following packages have unmet dependencies:
 libnginx-mod-http-passenger : Depends: nginx-common (< 1.14.1) but 1.15.5-0ubuntu2.1 is to be installed
E: Unable to correct problems, you have held broken packages.

**Comment by voordev** [comment]
> > @NSLog0 can you post the exact series of steps you took, and the output from `apt-get` that you're seeing?
> 
> I was following a guide posted on Medium (obviously that is not your guide, but the steps are about spinning up a Virtual Server on Digital Ocean -
> 
> https://medium.com/ruby-on-rails-web-application-development/how-to-deploy-ruby-on-rails-apps-to-the-internet-production-staging-49efc503c91d.
> 
> It looks like someone from your team mentioned the issue in a blog post here https://blog.phusion.nl/2018/10/03/passenger-5-3-5/
> 
> I am failing on this command
> 
> sudo apt-get install -y libnginx-mod-http-passenger
> 
> I get the failure message
> 
> "libnginx-mod-http-passenger : Depends: passenger (= 1:5.1.12-1~zesty1) but it is not going to be installed
> Depends: nginx-common (= 1.10.3-1ubuntu3.1) but 1.14.0-0ubuntu1.2 is to be installed"
> 
> In the guide I am following, it is the step that is underneath the "Web Server" header. It seems like from the comments in the blog post that it may not be an operating system issue. Hope this helps.
> 
> ## --Solution for me on Ubuntu 18.04 Digital Ocean Droplet--
> It seems to be related to the line
> 
> sudo sh -c 'echo deb https://oss-binaries.phusionpassenger.com/apt/passenger bionic main > /etc/apt/sources.list.d/passenger.list'
> 
> I was trying to run 'zesty' instead of 'bionic' (Bionic worked)
> 
> Follow the steps posted here -- https://www.phusionpassenger.com/library/install/nginx/install/oss/bionic/#step-1:-install-passenger-packages (also shown in the image below).
> 
> ![phusion_failingtoinstall_nginxdependency](https://user-images.githubusercontent.com/12903168/52007315-d5f66d00-2493-11e9-8bc7-74d7fa25bd70.PNG)

And this was the solution.
The docs are still incorrect, if you want people to use your product or even buy the premium make sure the damn documentation is up2date. sorry for the harsh words but this is juts plain stupid and cost me another 30 minutes down the drain!

**Comment by CamJN** [comment]
The maintained docs are at https://www.phusionpassenger.com/docs/advanced_guides/install_and_upgrade/apache/install/

**Comment by faisalraza** [comment]
@CamJN I am on Ubuntu 19.04 and facing the following issue on running the command 
`sudo apt-get install -y libnginx-mod-http-passenger`

```
The following packages have unmet dependencies:
libnginx-mod-http-passenger : Depends: nginx-common (< 1.14.1) but 1.15.9-0ubuntu1 is to be installed
E: Unable to correct problems, you have held broken packages.
```

I executed `sudo apt clean && sudo apt update` but still got the same issue on installing `libnginx-mod-http-passenger`.

Then I updated  "bionic" to "cosmic"  in `/etc/apt/sources.list.d/passenger.list` file and got the following issue on installing ibnginx-mod-http-passenger:

```Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package libnginx-mod-http-passenger```

Can you please assist how to fix it?

**Comment by CamJN** [comment]
@faisalraza our ubuntu 19.04 packages are not yet released due to being blocked on our migration to a new linux package host.

**Comment by AntonAL** [comment]
On Ubuntu `18.04.3 LTS` and with `nginx/1.17.6` I have:

```bash
$ sudo apt-get install -y libnginx-mod-http-passenger
…
The following packages have unmet dependencies:
 libnginx-mod-http-passenger : Depends: nginx-common (< 1.14.1) but 1.16.1-0+bionic1 is to be installed
```

**Comment by CamJN** [comment]
@AntonAL i just tested installing `libnginx-mod-http-passenger` on bionic in a clean vm, and it worked fine. Can you provide a script to go from a clean environment to reproduction of the issue?

**Comment by jfoclpf** [comment]
> The maintained docs are at https://www.phusionpassenger.com/docs/advanced_guides/install_and_upgrade/apache/install/

This solved my issue, thanks! Pople say we should change to cosmic or whatever but of course that depends on your versions, check it with `lsb_release -a` and then just follow the link

**Comment by jfoclpf** [comment]
this error happens after an upgrade of Ubuntu and passenger ubuntu release is not update, you can solve the problem by remove passenger and installing it again (save any config files you may have)

```
sudo apt purge passenger -y
sudo apt install passenger -y
sudo apt install libnginx-mod-http-passenger -y
```
